File: bone-blueprint/src/main/java/com/bone/blueprint/application/usecase/simple/GetOrderDetailUseCase.java
package com.bone.blueprint.application.usecase.simple;

import com.bone.blueprint.application.query.dto.OrderDto;
import com.bone.blueprint.application.query.handler.OrderDetailQueryHandler;
import com.bone.blueprint.application.query.qry.OrderDetailQuery;
import com.bone.core.usecase.UseCaseExecutor;
import com.bone.core.usecase.UseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@UseCase(
    name = "GetOrderDetail",
    description = "获取订单详情",
    transactional = false
)
@Service
@RequiredArgsConstructor
public class GetOrderDetailUseCase implements UseCaseExecutor<OrderDetailQuery, OrderDto> {

    private final OrderDetailQueryHandler orderDetailQueryHandler;

    @Override
    public OrderDto execute(OrderDetailQuery query) {
        return orderDetailQueryHandler.handle(query);
    }
}

File: bone-engine/go-engine/bone-blueprint-go/application/usecase/simple/get_order_detail_use_case.go
package simple

import (
	"context"

	"github.com/bone-engine/bone-blueprint-go/application/query/dto"
	"github.com/bone-engine/bone-blueprint-go/application/query/qry"
	"github.com/bone-engine/bone-blueprint-go/application/query/handler"
)

// GetOrderDetailUseCase 获取订单详情用例
type GetOrderDetailUseCase struct {
	detailQueryHandler *handler.OrderDetailQueryHandler
}

// NewGetOrderDetailUseCase 创建获取订单详情用例
func NewGetOrderDetailUseCase(detailQueryHandler *handler.OrderDetailQueryHandler) *GetOrderDetailUseCase {
	return &GetOrderDetailUseCase{
		detailQueryHandler: detailQueryHandler,
	}
}

// Execute 执行用例
func (uc *GetOrderDetailUseCase) Execute(ctx context.Context, orderID int64) (*dto.OrderDto, error) {
	query := &qry.OrderDetailQuery{OrderID: orderID}
	return uc.detailQueryHandler.Handle(ctx, query)
}
